packages/edit-site/src/components/sidebar-navigation-screen-pages/index.js

```diff
@@ -4,61 +4,150 @@
 import {
 	__experimentalItemGroup as ItemGroup,
 	__experimentalItem as Item,
+	__experimentalVStack as VStack,
 } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
-import { useEntityRecords } from '@wordpress/core-data';
+import { useEntityRecords, store as coreStore } from '@wordpress/core-data';
 import { decodeEntities } from '@wordpress/html-entities';
+import { layout, page, home, loop } from '@wordpress/icons';
+import { useSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import SidebarNavigationScreen from '../sidebar-navigation-screen';
 import { useLink } from '../routes/link';
 import SidebarNavigationItem from '../sidebar-navigation-item';
-import SidebarNavigationSubtitle from '../sidebar-navigation-subtitle';
 
-const PageItem = ( { postId, ...props } ) => {
+const PageItem = ( { postType = 'page', postId, ...props } ) => {
 	const linkInfo = useLink( {
-		postType: 'page',
+		postType,
 		postId,
 	} );
 	return <SidebarNavigationItem { ...linkInfo } { ...props } />;
 };
 
 export default function SidebarNavigationScreenPages() {
-	const { records: pages, isResolving: isLoading } = useEntityRecords(
+	const { records: pages, isResolving: isLoadingPages } = useEntityRecords(
 		'postType',
-		'page'
+		'page',
+		{
+			per_page: -1,
+		}
 	);
+	const { records: templates, isResolving: isLoadingTemplates } =
+		useEntityRecords( 'postType', 'wp_template', {
+			per_page: -1,
+		} );
+
+	const dynamicPageTemplates =
+		templates &&
+		templates.filter( ( template ) => {
+			return template.slug === '404' || template.slug === 'search';
+		} );
+
+	const homeTemplate =
```

Comment:
We could simplify this part for better readability with something like:
```
const dynamicPageTemplates = templates?.filter( ( { slug } ) =>
    [ '404', 'search' ].includes( slug )
);
const homeTemplate = templates?.find( ( { slug } ) =>
    [ 'front-page', 'home', 'index' ].includes( slug )
);

```

Reply:
We can do that for `dynamicPageTemplates`, but for `homeTemplate` we want to be absolutely sure that if `front-page` exists it gets returned, else if  `home` exists it gets returned, instead of returning whichever of the three happens to be first in the array. That's why I was so verbose about it 😄

Reply:
Would something like this also work to reduce the potential iterations?

```js
	const homeTemplate = templates?.find( ( { slug } ) => {
		if ( slug === 'front-page' ) {
			return true;
		}
		if ( slug === 'home' ) {
			return true;
		}
		if ( slug === 'index' ) {
			return true;
		}
		return false;
	} );
```